step3/dataStores/Storage.js

```diff
@@ -0,0 +1,48 @@
+const uuidv4 = require('uuid/v4');
+
+class Storage {
+
+	constructor () {
+		this.chat = {
+			messages: [],
+			users: []
+		}
+	}
+
+	saveMessage(sender, body) {
+		this.chat.messages.push({
+			sender: sender,
+			body: body,
+			timestamp: Date.now(),
+			id: uuidv4()});
+	}
+
+	getMessages() {
+		return this.chat.messages;
+	}
+
+	connect(username) {
+		const index = this.chat.users.indexOf(username);
```

Comment:
w es6 są chyba jakieś fajniejsze konstrukcje na zawiera się / nie zawiera się w liście

Reply:
Co do sprawdzenia to jest ;-) Ale z tego co ja rozumiem tutaj IWO od razu chce znać pozycję, żeby później ją łatwo usunąć.